Sources/NIOHTTPClient/HTTPHandler.swift

```diff
@@ -346,7 +376,14 @@ internal class TaskHandler<T: HTTPClientResponseDelegate>: ChannelInboundHandler
                 break
             default:
                 self.state = .body
-                self.delegate.didReceivePart(task: self.task, body)
+                let future = self.delegate.didReceivePart(task: self.task, eventLoop: context.eventLoop, body)
+                self.mayRead = false
+                future.whenComplete { _ in
```

Comment:
What is the future is failed? Do we still want to `read()`?

Reply:
good catch, I think we may want to fail here, thanks!